### PR TITLE
Feature/add network version to p2p

### PR DESCRIPTION
**Pull request by LukasGasior1**

### CODE REVIEW

File: core/src/main/java/com/radixdlt/p2p/transport/handshake/AuthHandshaker.java [hunk 1]
```diff
@@ -181,10 +186,18 @@ public Pair<byte[], AuthHandshakeResult> handleInitialMessage(ByteBuf data) {
...
       final var plaintext = ecKeyOps.eciesDecrypt(encryptedPayload, sizeBytes);
-      final var message = serialization.fromDson(plaintext, AuthInitiateMessage.class);
+      final AuthInitiateMessage message;
+      try {
+        message = serialization.fromDson(plaintext, AuthInitiateMessage.class);
+      } catch (DeserializeException ex) {
+        return Pair.of(
+            new byte[] {STATUS_ERROR},
+            AuthHandshakeResult.error(
+                "Auth handshake failed (possible network version mismatch)", Optional.empty()));
```
Comment (dhedey) [inline core/src/main/java/com/radixdlt/p2p/transport/handshake/AuthHandshaker.java:196]: If an Olympia node keeps trying to connect, will we get spammed logs? Do we block them based on IP?

**Review by dhedey — APPROVED** [review]

**Review by malexandru-rdx — APPROVED** [review]
LGTM!

File: core/src/main/java/com/radixdlt/p2p/transport/handshake/AuthInitiateMessage.java [hunk 2]
```diff
@@ -95,34 +95,47 @@ public final class AuthInitiateMessage extends BaseHandshakeMessage {
   @DsonOutput(DsonOutput.Output.ALL)
   private final HashCode nonce;
 
-  @JsonProperty("networkId")
+  @JsonProperty("nid")
   @DsonOutput(DsonOutput.Output.ALL)
-  private final int networkId;
+  private final byte networkId;
+
+  @JsonProperty("networkVersion")
```
Comment (jakrawcz-rdx) [inline core/src/main/java/com/radixdlt/p2p/transport/handshake/AuthInitiateMessage.java:102]: isn't this more of a "protocolVersion"? or am I misinterpreting?

**Review by jakrawcz-rdx — APPROVED** [review]
Not a terribly elegant "forced protocol change" but I understand we gotta do it, and the high-level tests are convincing enough 👍 

(please see my optional suggestions)

File: core/src/main/java/com/radixdlt/p2p/transport/handshake/AuthInitiateMessage.java (hunk 2 above)
Comment (LukasGasior1) [reply core/src/main/java/com/radixdlt/p2p/transport/handshake/AuthInitiateMessage.java:102]: hmh, I think of it as a higher-level network version, not just the protocol

File: core/src/main/java/com/radixdlt/p2p/transport/handshake/AuthHandshaker.java (hunk 1 above)
Comment (LukasGasior1) [reply core/src/main/java/com/radixdlt/p2p/transport/handshake/AuthHandshaker.java:196]: We shouldn't be getting spammed logs, the address (on the initiator side) should get blacklisted after a failed handshake

File: core/src/main/java/com/radixdlt/p2p/transport/handshake/AuthInitiateMessage.java [hunk 3]
```diff
@@ -76,7 +76,7 @@
 import java.util.Objects;
 import java.util.Set;
 
-@SerializerId2("message.handshake.auth_initiate")
+@SerializerId2("message.handshake.v2.auth_initiate")
```
[the remark this answers is not in the document]
Comment (LukasGasior1) [reply core/src/main/java/com/radixdlt/p2p/transport/handshake/AuthInitiateMessage.java:79]: maybe just `networkId2` or `babylonNetworkId` to make the intention clear? 😄 and an accompanying comment

[the remark this answers is not in the document]
Comment (jakrawcz-rdx) [reply core/src/main/java/com/radixdlt/p2p/transport/handshake/AuthInitiateMessage.java:79]: `babylonNetworkId` is perfect 👍

**Pull request merged**